include/range/v3/utility/random.hpp

```diff
@@ -495,13 +495,31 @@ namespace ranges
             }
 
             using default_random_engine =
-                meta::if_c<sizeof(void*) >= 8, std::mt19937_64, std::mt19937>;
-            inline default_random_engine& get_random_engine()
+                meta::if_c<(sizeof(void*) >= 8), std::mt19937_64, std::mt19937>;
+
+            template <typename T = void>
+            default_random_engine& get_random_engine()
             {
+#ifdef TARGET_OS_IPHONE
+                static_assert(meta::invoke<meta::id<std::false_type>, T>::value,
+                    "IPhone did not support C++11 thread_local when I wrote this. "
+                    "If it does now, or you want to contribute a workourand, then "
+                    "by all means contribute a patch / complain in the range-v3 "
+                    "forum. Otherwise, you should avoid calling functions that "
+                    "accept an URNG without explicitly specifying an URNG.");
+                std::terminate();
```

Comment:
My preference in this case would be to have a single global engine and synchronize access to it. Unceremoniously dumping core is not a nice thing to do.